flocker/testtools/__init__.py

```diff
@@ -192,33 +194,61 @@ def assert_not_equal_comparison(case, a, b):
             "Expected a and b to be not-equal: " + "; ".join(messages))
 
 
-def loop_until(predicate):
+def function_serializer(function):
+    try:
+        return {
+            "function": str(function),
+            "file": getfile(function),
+            "line": getsourcelines(function)[1]
+        }
+    except IOError:
+        # One debugging method involves changing .py files and is incompatible
+        # with inspecting the source.
+        return {
+            "function": function,
+        }
+
+LOOP_UNTIL_ACTION = ActionType(
+    action_type="flocker:testtools:loop_until",
+    startFields=[Field("predicate", function_serializer)],
+    successFields=[Field("result", serializer=safe_repr)],
+    description="Looping until predicate is true.")
+
+LOOP_UNTIL_ITERATION_MESSAGE = MessageType(
+    message_type="flocker:testtools:loop_until:iteration",
+    fields=[Field("result", serializer=safe_repr)],
```

Comment:
This message should contain the predicate as well. See http://build.clusterhq.com/builders/flocker%2Facceptance%2Faws%2Fubuntu-14.04%2Faws/builds/612/steps/trial/logs/test.log and search for task `96e578bd-922b-4a39-bb30-71bdbe829ead` as an example where the `LOOP_UNTIL_ACTION` message is absent, presumably because logging is setup after the `loop_until` is entered.